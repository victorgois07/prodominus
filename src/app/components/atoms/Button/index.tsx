import { ButtonHTMLAttributes, forwardRef } from "react";
import { twMerge } from "tailwind-merge";

export const Button = forwardRef<
  HTMLButtonElement,
  ButtonHTMLAttributes<HTMLButtonElement>
>(({ className, ...props }, ref) => (
  <button
    ref={ref}
    className={twMerge(
      "px-4 py-2 rounded font-semibold transition disabled:opacity-50",
      className
    )}
    {...props}
  />
));
Button.displayName = "Button";
