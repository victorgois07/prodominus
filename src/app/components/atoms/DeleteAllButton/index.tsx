"use client";

import { useProducts } from "@/business/hooks/useProducts";
import { useConfirmDialog } from "@/business/store/useConfirmDialog";

export function DeleteAllButton() {
  const { deleteAllProducts } = useProducts();
  const { show } = useConfirmDialog();

  const handleDeleteAll = () => {
    show(
      "Tem certeza que deseja excluir todos os produtos? Esta ação não pode ser desfeita.",
      async () => {
        await deleteAllProducts();
      }
    );
  };

  return (
    <button
      onClick={handleDeleteAll}
      className="bg-red-600 hover:bg-red-700 text-white px-4 py-2 rounded-md transition-colors"
    >
      Excluir Todos
    </button>
  );
}
