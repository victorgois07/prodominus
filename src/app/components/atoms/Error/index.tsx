interface ErrorProps {
  message: string;
}

export function Error({ message }: ErrorProps) {
  return (
    <div className="flex items-center justify-center min-h-[400px]">
      <div
        className="bg-red-50 border border-red-200 text-red-700 px-4 py-3 rounded relative"
        role="alert"
      >
        <strong className="font-bold">Error: </strong>
        <span className="block sm:inline">{message}</span>
      </div>
    </div>
  );
}
