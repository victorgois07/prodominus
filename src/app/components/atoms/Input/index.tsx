import { InputHTMLAttributes, forwardRef } from "react";
import { twMerge } from "tailwind-merge";

export const Input = forwardRef<
  HTMLInputElement,
  InputHTMLAttributes<HTMLInputElement>
>(({ className, ...props }, ref) => (
  <input
    ref={ref}
    className={twMerge(
      "shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline",
      className
    )}
    {...props}
  />
));
Input.displayName = "Input";
