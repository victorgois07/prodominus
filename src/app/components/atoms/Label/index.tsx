import { LabelHTMLAttributes, forwardRef } from "react";

export const Label = forwardRef<
  HTMLLabelElement,
  LabelHTMLAttributes<HTMLLabelElement>
>((props, ref) => (
  <label ref={ref} className="block mb-1 font-medium" {...props} />
));
Label.displayName = "Label";
