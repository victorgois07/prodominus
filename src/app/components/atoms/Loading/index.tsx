export function Loading() {
  return (
    <div className="flex items-center justify-center min-h-[400px]">
      <div className="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-green-500"></div>
    </div>
  );
}
