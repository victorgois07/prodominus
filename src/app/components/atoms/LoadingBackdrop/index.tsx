"use client";

import { useLoadingBackdrop } from "@/business/store/useLoadingBackdrop";

export function LoadingBackdrop() {
  const { isOpen } = useLoadingBackdrop();

  if (!isOpen) return null;

  return (
    <div className="fixed inset-0 bg-black/50 backdrop-blur-sm z-50 flex items-center justify-center">
      <div
        role="status"
        className="animate-spin rounded-full h-16 w-16 border-t-2 border-b-2 border-white"
      ></div>
    </div>
  );
}
