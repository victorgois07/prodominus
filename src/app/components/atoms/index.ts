export * from "./Button";
export * from "./Error";
export * from "./Input";
export * from "./Label";
export * from "./Loading";
export * from "./Toast";
