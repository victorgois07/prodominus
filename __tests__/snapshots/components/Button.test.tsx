import { Button } from "@/app/components/atoms/Button";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

describe("Button Snapshot", () => {
  it("should match snapshot with primary variant", () => {
    const { container } = render(
      <Button variant="primary">Primary Button</Button>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with secondary variant", () => {
    const { container } = render(
      <Button variant="secondary">Secondary Button</Button>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with danger variant", () => {
    const { container } = render(
      <Button variant="danger">Danger Button</Button>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with loading state", () => {
    const { container } = render(
      <Button variant="primary" loading>
        Loading Button
      </Button>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with disabled state", () => {
    const { container } = render(
      <Button variant="primary" disabled>
        Disabled Button
      </Button>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with icon", () => {
    const { container } = render(
      <Button variant="primary" icon="plus">
        Button with Icon
      </Button>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with long text", () => {
    const { container } = render(
      <Button variant="primary">
        This is a very long button text that should be properly displayed
      </Button>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with small size", () => {
    const { container } = render(
      <Button variant="primary" size="sm">
        Small Button
      </Button>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with large size", () => {
    const { container } = render(
      <Button variant="primary" size="lg">
        Large Button
      </Button>
    );

    expect(container).toMatchSnapshot();
  });
});
