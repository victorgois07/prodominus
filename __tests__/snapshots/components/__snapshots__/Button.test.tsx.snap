// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Snapshot > should match snapshot with danger variant 1`] = `
<div>
  <button
    class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
    variant="danger"
  >
    Danger Button
  </button>
</div>
`;

exports[`Button Snapshot > should match snapshot with disabled state 1`] = `
<div>
  <button
    class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
    disabled=""
    variant="primary"
  >
    Disabled Button
  </button>
</div>
`;

exports[`Button Snapshot > should match snapshot with icon 1`] = `
<div>
  <button
    class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
    icon="plus"
    variant="primary"
  >
    Button with Icon
  </button>
</div>
`;

exports[`Button Snapshot > should match snapshot with large size 1`] = `
<div>
  <button
    class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
    variant="primary"
  >
    Large Button
  </button>
</div>
`;

exports[`Button Snapshot > should match snapshot with loading state 1`] = `
<div>
  <button
    class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
    variant="primary"
  >
    Loading Button
  </button>
</div>
`;

exports[`Button Snapshot > should match snapshot with long text 1`] = `
<div>
  <button
    class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
    variant="primary"
  >
    This is a very long button text that should be properly displayed
  </button>
</div>
`;

exports[`Button Snapshot > should match snapshot with primary variant 1`] = `
<div>
  <button
    class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
    variant="primary"
  >
    Primary Button
  </button>
</div>
`;

exports[`Button Snapshot > should match snapshot with secondary variant 1`] = `
<div>
  <button
    class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
    variant="secondary"
  >
    Secondary Button
  </button>
</div>
`;

exports[`Button Snapshot > should match snapshot with small size 1`] = `
<div>
  <button
    class="px-4 py-2 rounded font-semibold transition disabled:opacity-50"
    variant="primary"
  >
    Small Button
  </button>
</div>
`;
